src/stages/stage2/timeline.c

```diff
@@ -80,16 +80,9 @@ TASK(spinshot_fairy_attack, {
 		)
 	);
 
-	cmplx ref_pos;
-	Enemy *e;
-
 	for(int t = 0; t < ARGS.activate_time; ++t, YIELD) {
 		int live_count = 0;
-		e = ENT_UNBOX(ARGS.e);
-
-		if(e) {
-			ref_pos = e->pos;
-		}
+		cmplx ref_pos = NOT_NULL(ENT_UNBOX(ARGS.e))->pos;
```

Comment:
Not like that. The code inside the `for` loop must not assume that `ENT_UNBOX(ARGS.e)` isn't NULL, it has to check. It is guaranteed to not be null only on the very first iteration (after that we `YIELD`, and the entity may die before the task is re-entered; this is a valid state). GCC doesn't know that the first `ENT_UNBOX` is a guaranteed non-null, hence the maybe-uninitialized warning. 

Here's what needs to be done:

* Code inside the loop has to be reverted to its pre-PR state; it is correct.
* `cmplx ref_pos = NOT_NULL(ENT_UNBOX(ARGS.e))->pos` needs to go before the loop.
    * Technically `ref_pos` could be initialized to anything, e.g. 0. `NOT_NULL(ENT_UNBOX(ARGS.e))->pos` just adds an extra sanity check (to ensure that we didn't call the task with an invalid `ARGS.e` in the first place), and establishes that this variable is meant to track position of `ARGS.e`.